packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -611,11 +614,22 @@ class DeviceDomain extends Domain {
 
   /// Return a list of the current devices, with each device represented as a map
   /// of properties (id, name, platform, ...).
+  ///
+  /// Optionally takes a directory path, which will filter down the list of
+  /// devices to only those supported by the current project.
   Future<List<Map<String, dynamic>>> getDevices([ Map<String, dynamic> args ]) async {
     final List<Map<String, dynamic>> devicesInfo = <Map<String, dynamic>>[];
 
+    FlutterProject flutterProject;
+    if (args.containsKey('projectDirectory')) {
```

Comment:
we should update the daemon.md spec doc to mention the optional `projectDirectory` field